contrib/olivere/elastic/elastictrace.go

```diff
@@ -1,86 +1,120 @@
-// Package elastictrace provides tracing for the Elastic Elasticsearch client.
-// Supports v3 (gopkg.in/olivere/elastic.v3), v5 (gopkg.in/olivere/elastic.v5)
-// but with v3 you must use `DoC` on all requests to capture the request context.
-package elastictrace
+// Package elastic provides functions to trace the gopkg.in/olivere/elastic.v{3,5} packages.
+package elastic
 
 import (
 	"bytes"
 	"errors"
+	"fmt"
 	"io/ioutil"
+	"net"
 	"net/http"
+	"regexp"
 	"strconv"
+	"time"
 
 	"github.com/DataDog/dd-trace-go/tracer"
 	"github.com/DataDog/dd-trace-go/tracer/ext"
 )
 
-// MaxContentLength is the maximum content length for which we'll read and capture
-// the contents of the request body. Anything larger will still be traced but the
-// body will not be captured as trace metadata.
-const MaxContentLength = 500 * 1024
+// NewHTTPClient returns a new http.Client which traces requests under the given service name.
+//
+// TODO(gbbr): Remove tracer argument when we switch to OpenTracing.
+func NewHTTPClient(service string, tracer *tracer.Tracer) *http.Client {
```

Comment:
Don't we provide something like other integrations, where you don't need to set the `service` and the `tracer`? (so a `NewHTTPClient` without params that automatically sets some defaults).

Reply:
We do that to match the signature of the original method or function. In this case, the function is completely new so there is no need to. We could create a default though. The tracer will be removed anyway in the next PR.

Reply:
OK since in this release I want to keep consistency, let's do a globale docs review before releasing 0.7.0. Checking one by one all examples, we can notice differences / incoherences and in that case we can alter this new API. The rest is fine, so we can keep this one since it's a new method.